CA_DataUploaderLib/MCUBoard.cs

```diff
@@ -410,13 +414,17 @@ private async Task<string> ReadLine()
                     if (res.IsCanceled)
                         throw new TimeoutException("timed out (soft)");
                     var buffer = res.Buffer;
-                    var readLine = TryReadLine(ref buffer, out var line);
-                    pipeReader.AdvanceTo(buffer.Start, buffer.End); // Tell the PipeReader how much of the buffer has been consumed.
+                    var readLine = tryReadLine(ref buffer, out var line);
```

Comment:
If tryReadLine throws the pipeReader is not advanced / this could potentially cause a tight loop if some custom read line parser throws exceptions. Note that parsers throwing exceptions must be avoided anyway as it prevents making progress on the data, nevertheless the tight loop would make the bad situation worse.

https://docs.microsoft.com/en-us/dotnet/standard/io/pipelines
there is a reference ReadSingleMessageAsync that shows an example avoiding the tight loop